File: type-script-nest-graph-ql-실습/src/app/supply/model/dto/response/supply-response.dto.ts
import { SupplyEntity } from "../../entities/supply.entity";

export class SupplyResponseDto {
  private supplyId: string;
  private teamId: number;
  constructor(supply: SupplyEntity) {
    this.supplyId = supply.supplyId;
    this.teamId = supply.team.teamId;
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/supply/model/dto/response/SupplyResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.supply.model.dto.response

import com.junyharangstudy.kotlingraphqltest.api.supply.model.entity.Supply

class SupplyResponseDto (
    private val supplyId: String,
    private val teamId: Long
) {
    companion object {
        fun supplyToDto(supply: Supply): SupplyResponseDto {
            return SupplyResponseDto(
                supplyId = supply.supplyId,
                teamId = supply.team.teamId!!
            )
        }
    }
}